src/dex-ui/pages/ProposalDetails/useProposalDetails.tsx

```diff
@@ -29,6 +30,16 @@ export function useProposalDetails(proposalId: string | undefined) {
 
   const isLoadingDialogOpen =
     executeProposal.isLoading || castVote.isLoading || claimGODTokens.isLoading || cancelProposal.isLoading;
+
+  let isWalletConnected = false;
+  try {
+    isWalletConnected = wallet.getSigner().getAccountId().toString().length !== 0;
```

Comment:
The wallet connection state is determined by the `HashConnectConnectionState` returned from Hashpack. The `wallet.hashConnectConnectionState` field stores the value returned from Hashpack.

```
export declare enum HashConnectConnectionState {
    Connecting = "Connecting",
    Connected = "Connected",
    Disconnected = "Disconnected",
    Paired = "Paired"
}
```

If `wallet.hashConnectConnectionState === HashConnectConnectionState.Paired`, then the wallet is considered fully paired and operational.

You could create a `wallet.isPaired` function to standardize this logic.

Reply:
done, I have put `isPaired` and `GODTokenBalance` in wallet store for simplicity of the logic.